### DISCUSSION TITLE
"let instance = await..." binds not the contract being called, but called "TruffleContract".

### DISCUSSION THREAD

**Discussion by Ferelur** [ConsenSys-archive/truffle#6000] [General Q&A]
I decided to start learning truffle using the documentation and quick start instructions posted on the site, but at the time of using the command in the console
`let instance = await MetaCoin.deployed()`
the console output "undefined", and writes to the instance TruffleContract.
How can this be fixed?
P.S. I'm sorry if I didn't attach important data, this is my first such request.

**Comment by FlaviuWeb3** [comment]
It sounds like you may be encountering an issue with the deployment of the MetaCoin contract using Truffle. Here are a few things you can try to fix the issue:

1. Make sure that you have successfully compiled your contracts using the truffle compile command before running the deployed() function. If the contract has not been compiled successfully, the deployed() function may return undefined.

2. Check that you have defined a contract named MetaCoin in your Truffle project. You can do this by looking at the contracts/ directory in your Truffle project and verifying that there is a contract file named MetaCoin.sol. If the contract file does not exist, you will need to create it before attempting to deploy it.

3. Try restarting your Truffle development console and re-deploying the contract using the truffle migrate command. After the contract has been successfully deployed, you can try running the MetaCoin.deployed() command again.

**Reply by cds-amal** [reply]
This is NODE REPL behavior and doesn't imply there was a deployment issue. See this [link for an explanation](https://blog.bitsrc.io/why-does-console-log-return-undefined-e06d44b4d0f8).

**Comment by cds-amal** [comment]
> I decided to start learning truffle using the documentation and quick start instructions posted on the site, but at the time of using the command in the console `let instance = await MetaCoin.deployed()` the console output "undefined", and writes to the instance TruffleContract. How can this be fixed? P.S. I'm sorry if I didn't attach important data, this is my first such request.

This is not an issue with deployment, but rather [how REPLs work](https://blog.bitsrc.io/why-does-console-log-return-undefined-e06d44b4d0f8). If you were to examine instance, you'd see it was bound to the deployed contract instance. 

```console
Summary
=======
> Total deployments:   2
> Final cost:          0.001894367934056634 ETH


truffle(develop)> let instance = await MetaCoin.deployed()
undefined
truffle(develop)> instance.abi.length
5
truffle(develop)> // This shows that instance has content, and the deployment did in fact succeed.
```
